Search by Category: unit tests

SaiShan opened the pull request:
User Story: #54 
Task: #114

SaiShan committed:
unit tests

SaiShan committed:
cleaning code

OlivierTD reviewed (changes requested):
Please review your tests thanks!

SaiShan committed:
Merge branch 'master' of https://github.com/OlivierTD/team-rocket into searchByCat

SaiShan committed:
Merge branch 'master' of https://github.com/OlivierTD/team-rocket into searchByCat

SaiShan committed:
fixing one test

SaiShan committed:
Merge branch 'searchByCat' of https://github.com/OlivierTD/team-rocket into searchByCat

SaiShan committed:
unit test for search by category

testing the builder, if the alert dialog contains the appropriate values as well as if the newly created fragment is properly set

SaiShan committed:
Merge branch 'master' of https://github.com/OlivierTD/team-rocket into searchByCat

SaiShan committed:
fixing naming error for testHomeFragment

davidlegare reviewed (changes requested):
You can't mock the fragment that you are testing, it beats the whole purpose of testing it, you just mock the dependencies.

davidlegare commented:
There are conflicts to resolve as well.

SaiShan committed:
unmock tested fragment

SaiShan committed:
resolving merge issues

SaiShan commented on app/src/test/java/categorySearchTest/CategorySearchTest.java:
```diff
@@ -0,0 +1,118 @@
+
+    //testing the creation of an alert dialog
+    @Test
+    public void builderTest() throws Exception{
```
From my research, this is the only kind of test that i have found for alert dialog. Other functions for this user story, such as the search, have the same issues as the centralized search.

SaiShan committed:
Merge branch 'master' into searchByCat
